core/geometry/src/clipping/ClipPrimitive.ts

```diff
@@ -425,8 +425,10 @@ export class ClipShape extends ClipPrimitive {
     if (polygon.length < 3)
       return undefined;
     const pPoints = polygon.slice(0);
-    // Add closure point
-    if (!pPoints[0].isExactEqual(pPoints[pPoints.length - 1]))
+    // Add closure point.
+    if (pPoints[0].isAlmostEqual(pPoints[pPoints.length - 1]))
+      pPoints[0].clone(pPoints[pPoints.length - 1]);
+    else
       pPoints.push(pPoints[0]);

```

Comment:
Should the else case be cloning the point...otherwise why does the if case clone?

Reply:
Yes, it should be. Earlin fixed that in his PR on master.